### PR TITLE
Implement `list_logged_model_artifacts`

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/13340?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13340/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13340
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

```python
import tempfile
from pathlib import Path
import mlflow

from contextlib import contextmanager


@contextmanager
def show_request_and_response():
    from requests import Session
    from unittest import mock

    original = Session.request

    def patch(self, *args, **kwargs):
        res = original(self, *args, **kwargs)
        print("=== Request ===")
        print(args, kwargs)
        print("=== Response ===")
        print(res.text)
        return res

    with mock.patch.object(Session, "request", patch):
        yield


mlflow.set_tracking_uri("databricks")
mlflow.set_experiment("/Users/harutaka.kawamura@databricks.com/mlflow-3-artifacts")

with mlflow.start_run() as run:
    mlflow.log_artifact(__file__)
    mlflow.log_artifact(__file__, "model")

    client = mlflow.MlflowClient()
    print(client.list_artifacts(run.info.run_id))
    print(client.list_artifacts(run.info.run_id, "model"))

    model = client.create_logged_model(run.info.experiment_id, name="model")
    model = client.get_logged_model(model.model_id)
    with show_request_and_response():
        repo = mlflow.artifacts.get_artifact_repository(model.artifact_location)
        repo.log_artifact(__file__, "model")
        print(client.list_logged_model_artifacts(model.model_id))
        print(client.list_logged_model_artifacts(model.model_id, "model"))
```

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/store/artifact/artifact_repo.py [hunk 1]
```diff
@@ -145,6 +145,19 @@ def list_artifacts(self, path):
             List of artifacts as FileInfo listed directly under path.
         """
 
+    def list_logged_model_artifacts(self, path: Optional[str] = None) -> List[FileInfo]:
```
Comment (dbczumar) [inline mlflow/store/artifact/artifact_repo.py:148]: In databricks artifact repo, we didn't define separate methods for uploading / downloading artifacts. Should we be consistent for listing as well?

File: mlflow/store/artifact/databricks_artifact_repo_resources.py [hunk 2]
```diff
@@ -41,6 +60,18 @@ def __init__(self, id_: str, call_endpoint: Callable[..., Any]):
         self.id = id_
         self.call_endpoint = call_endpoint
         self.artifact_root = self.get_artifact_root()
+        self.relative_path = self.get_rel_path
+
+    def get_relative_path(self) -> str:
+        # Fetch the artifact root for the MLflow resource associated with `artifact_uri` and compute
+        # the path of `artifact_uri` relative to the MLflow Run's artifact root
```
Comment (dbczumar) [inline mlflow/store/artifact/databricks_artifact_repo_resources.py:67]:
```suggestion
        # the path of `artifact_uri` relative to the MLflow resource's artifact root
```

Do we need to update any similar inline comments?

File: mlflow/store/artifact/artifact_repo.py (hunk 1 above)
Comment (harupy) [reply mlflow/store/artifact/artifact_repo.py:148]: Ok, we can just remove this method and use list_artifacts?

File: mlflow/tracking/_tracking_service/client.py [hunk 3]
```diff
@@ -873,6 +874,20 @@ def _get_artifact_repo(self, run_id):
...
+
+        logged_model = self.get_logged_model(model_id)
+        artifact_uri = add_databricks_profile_info_to_artifact_uri(
+            logged_model.info.artifact_location, self.tracking_uri
+        )
+        artifact_repo = get_artifact_repository(artifact_uri)
+        if len(utils._artifact_repos_cache) > 1024:
+            utils._artifact_repos_cache.popitem(last=False)
+        utils._artifact_repos_cache[model_id] = artifact_repo
+        return artifact_repo
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/tracking/_tracking_service/client.py]: I removed `_get_artifact_repo_for_logged_model` and updated `_get_artifact_repo` to reuse it.

**Pull request merged**